File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long l= long(INT_MAX);
        string s=to_string(x);
        int i=0,j=s.size()-1;

        while(i<j)
        {
            swap(s[i],s[j]);
            i++;
            j--;
        }

        //cout <<s<<endl;

        long t= stol(s);

        if(s[s.size()-1]=='-')
        {
            t=t*(-1);
        }

        if (l<t || long(INT_MIN)>t)
            return 0;
        
        return int(t);
    }
};

File: 7-reverse-integer/reverse-integer.py
import sys

class Solution:
    def reverse(self, x: int) -> int:
        b=False
        if x < 0:
            b=True

        s = str(abs(x))
        s=s[::-1]
        if int(s) > 2147483647:
            return 0
        if b:
            if (-1*int(s)) < -2147483648:
                return 0
            else:
                return (-1*int(s))
        
        return int(s)
        